rfcs/process.md

```diff
@@ -0,0 +1,235 @@
+
+## When you need to follow this process
+
+You need to follow this process if you intend to make "substantial"
+changes to the test262 suite and harness.
+What constitutes a "substantial" change is evolving based on community
+norms, but may include the following.
+
+- Additions or changes to [INTERPRETING.md](https://github.com/tc39/test262/blob/main/INTERPRETING.md) that require consumers
+  of the test suite to make changes to their test runners.
+- Additions to [features.txt](https://github.com/tc39/test262/blob/main/features.txt) that are not new Stage 3 proposals,
+  or removals of feature flags.
+- Substantial changes and new features in the [harness files](https://github.com/tc39/test262/tree/main/harness).
+
+Some changes do not require an RFC:
+
+- Adding test coverage.
+- Bug fixes.
+- Adding features to [features.txt](https://github.com/tc39/test262/blob/main/features.txt) that are new Stage 3 proposals.
+- Bug fixes to the [harness files](https://github.com/tc39/test262/tree/main/harness).
+
+If you submit a pull request for a "substantial" change without going
+through the RFC process, it may be closed with a polite request to
+submit an RFC first.
+
+## Before creating an RFC
+
+Laying some groundwork ahead of the RFC can make the process smoother.
+Getting people to agree with an RFC often requires some effort towards
+consensus-building.
+
+The most common preparations for writing and submitting an RFC include
+talking the idea over on our [Matrix channel](https://matrix.to/#/#test262-maintainers:matrix.org), discussing the topic
+on our [discussion forum](https://es.discourse.group/c/test262/FIXME), and occasionally posting "pre-RFCs" on
+the forum.
+
+As a rule of thumb, receiving encouraging feedback from consumers of the
+test suite or the maintainers is a good indication that the RFC is worth
+pursuing.
+
+## What the process is
+
+In short, the process is to open a pull request with the RFC text, and
+get it merged into the test262 repository as a markdown file.
+At that point the RFC is "active" and may be implemented with the goal
+of eventual inclusion into test262.
+
+- Fork the test262 repository.
+
+- Copy `rfc/0000-template.md` to `rfc/0000-my-feature.md` (where
+  "my-feature" is the description of your RFC).
+  Don't assign an RFC number yet; this is going to be the PR number and
+  we'll rename the file accordingly once the PR is opened.
+
+- Fill in the RFC.
+  Put care into the details: RFCs that do not present convincing
+  motivation, demonstrate lack of understanding of the design's impact,
+  or are disingenuous about the drawbacks or alternatives tend to be
+  poorly-received.
+
+- Submit a pull request.
+  As a pull request the RFC will receive design feedback from the
+  various stakeholders, and the author should be prepared to revise it
+  in response.
+
+- Now that your RFC has an open pull request, use the issue number of
+  the PR to update your 0000- prefix to that number.
```

Comment:
i've seen a few places use this convention, and it seems like it adds unnecessary churn. is there a benefit to using the number in the filename that git history doesn't provide?

Reply:
I'd find that the numbering causes a lot of unnecessary conflicts too.

Reply:
Having the number in the filename seems like a benefit if you want to be able to colloquially refer to "RFC 1234" rather than "the deepEqual RFC". Otherwise you'll have to look for your title in the crowded `rfcs/` directory. That said, this part comes from Rust and I'm not attached to it.